src/components/SearchBar.jsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react'
+
+function SearchBar({product, onSearch,  onStockChange}) {
+
+ const handleSearchChange = (event) => {
+  const {value} = event.target;
+  onSearch(value)
+ }
+
+ const handleStockChange = (event) => {
+  onStockChange(event.target.checked);
+};
+
+  return (
+    
+    <>
+    <p>Search</p>
+    <form className="row g-3">
+        
+        <div className="col-auto">
+        <input 
+        className="form-control" 
+        placeholder="search"
+        onChange={handleSearchChange}
+        />
+       </div>
+        
+        <div className="col-auto">
+          <label>
+            <input type="checkbox" onChange={handleStockChange} />
+            Show only products in stock
+          </label>
+        </div>
+    </form>
+    </>
```

Comment:
```suggestion
    <>
      <p>Search</p>
      <form className="row g-3">
          
          <div className="col-auto">
          <input 
          className="form-control" 
          placeholder="search"
          onChange={handleSearchChange}
          />
         </div>
          
          <div className="col-auto">
            <label>
              <input type="checkbox" onChange={handleStockChange} />
              Show only products in stock
            </label>
          </div>
      </form>
    </>
```
Indentamelo por dios!!!